selfdrive/common/messagehelp.h

```diff
@@ -40,3 +40,57 @@ class MessageReader {
   capnp::FlatArrayMessageReader *msg_reader;
   cereal::Event::Reader event;
 };
+
+#define STACK_SEGEMENT_BUF_SIZE 256
+class MessageBuilder : public capnp::MessageBuilder {
+ public:
+  MessageBuilder(uint firstMallocSize = 2048) : returnedFirstSegment(false), nextSize(firstMallocSize), stackSegment{} {};
```

Comment:
What is returnedFirstSegment? Is it ever true? Maybe add a comment

Reply:
this variable is used to make sure stack will only be returned once at first segment.

Reply:
and why does there have to be a firstMallocSize?

I'm not sure these perf improvements are worth the hit to readability. They can always be made later in one place if a benchmark shows they are good.

Reply:
firstMallocSize is used to avoid too many alloc calls & segments for large message such as proclogd.for example ,MallocMessageBuilder's  default behaviors is alloc 1024 words everytime .so ,it need's alloc many segments to perpare the prologd's message.

Reply:
yes you'r right.I'm not sure isn't it worth too.so I tuned this PR to draft,close it later.

Reply:
not only 1 malloc.  save 2 malloc and one copy for a single message.

Reply:
I like the lines this PR removes from the programs. I still think it's worth it, just with the simplest implementation in the class.